### PR TITLE
fix(TDI-43092): eliminate region ID

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-43092

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/244/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/244/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/249/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/249/)

File: components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java [hunk 1]
```diff
@@ -412,12 +397,21 @@ public boolean postDeserialize(int version, PostDeserializeSetup setup, boolean
             loginTimeout.setValue(DEFAULT_LOGIN_TIMEOUT);
             migrated = true;
         }
+        if (version < 2 && region.getValue() != null) {
+            if (useCustomRegion.getValue() && customRegionID.getValue() != null) {
```
Comment (wwang-talend) [inline components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: getValue is the real value when context case, for example :
context.region is null in customer job, then the new regionID field will be empty and not context.region.

Comment (wwang-talend) [reply components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: here advice to use customRegionID. getStoredValue()

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/251/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/251/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/262/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/262/)

**Pull request merged**